swim/state_transitions.go

```diff
@@ -46,30 +47,40 @@ type transitionTimer struct {
 type stateTransitions struct {
 	sync.Mutex
 
-	node *Node
-
+	node     *Node
+	clock    clock.Clock
```

Comment:
In my mind, time is a global. Since we don't have control over the time package we need to store this "global" somewhere, which we did in the node. We never want the clocks on both instances to have different values. I think the coupling is very much needed, not unnecessary. Therefore, for me, it makes more sense to have one clock. Choose what you like though.